fastly/image_optimizer_default_settings_test.go

```diff
@@ -0,0 +1,290 @@
+package fastly
+
+import (
+	"strings"
+	"testing"
+)
+
+// TestClient_ImageOptimizerDefaultSettings tests the Image Optimizer Default Settings API
+//
+// NOTE: to run this, all backends on the active version of the test service must have shielding
+// enabled, and the test service's account must be allowed to enable Image Optimization.
+func TestClient_ImageOptimizerDefaultSettings(t *testing.T) {
+	t.Parallel()
+
+	fixtureBase := "image_optimizer_default_settings/"
+
+	testVersion := createTestVersion(t, fixtureBase+"version", testServiceID)
+
+	var err error
+
+	// Enable IO
+	record(t, fixtureBase+"enable_product", func(c *Client) {
+		_, err = c.EnableProduct(&ProductEnablementInput{
+			ProductID: ProductImageOptimizer,
+			ServiceID: testServiceID,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure we disable IO on the service after the test
+	defer func() {
+		record(t, fixtureBase+"disable_product", func(c *Client) {
+			err = c.DisableProduct(&ProductEnablementInput{
+				ProductID: ProductImageOptimizer,
+				ServiceID: testServiceID,
+			})
+		})
+
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	var defaultSettings *ImageOptimizerDefaultSettings
+
+	// Fetch
+	record(t, fixtureBase+"original_fetch", func(c *Client) {
+		defaultSettings, err = c.GetImageOptimizerDefaultSettings(&GetImageOptimizerDefaultSettingsInput{
+			ServiceID:      testServiceID,
+			ServiceVersion: *testVersion.Number,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	originalSettings := defaultSettings
+
+	// Reset our settings back to the original
+	defer func() {
+		record(t, fixtureBase+"final_reset", func(c *Client) {
+			if originalSettings != nil {
+				_, err = c.UpdateImageOptimizerDefaultSettings(&UpdateImageOptimizerDefaultSettingsInput{
+					ServiceID:      testServiceID,
+					ServiceVersion: *testVersion.Number,
+					// ResizeFilter:   &originalSettings.ResizeFilter,
+					Webp:        &originalSettings.Webp,
+					WebpQuality: &originalSettings.WebpQuality,
+					// JpegType:       &originalSettings.JpegType,
```

Comment:
Are these two comments meant to be deleted?

Reply:
Fixed!